tools/batchTestGenerator/cmd/root.go

```diff
@@ -69,20 +67,18 @@ func newCommandConfig() *commandConfig {
 			c.runConfig.TestCaseFilePath = filepath.Join(defaultFileLoc, "testcases.json")
 		}
 
-		eks64vars, err := transformEKSvars(c.eksARM64Flags)
-		if err != nil {
-			return fmt.Errorf("failed to transform eks arm 64 vars: %w", err)
+		if c.runConfig.MaxBatches < 1 {
+			return fmt.Errorf("max batches must be greater than 0")
 		}
-		c.runConfig.EksARM64Vars = eks64vars
 
-		eksVars, err := transformEKSvars(c.eksFlags)
+		err = c.runConfig.UnmarshalInputFile()
 		if err != nil {
-			return fmt.Errorf("failed to transform eks vars: %w", err)
+			return fmt.Errorf("error parsing input file: %w", err)
 		}
-		c.runConfig.EksVars = eksVars
 
-		if c.runConfig.MaxBatches < 1 {
-			return fmt.Errorf("max batches must be greater than 0")
+		err = c.runConfig.ValidateTestCaseInput()
+		if err != nil {
+			return fmt.Errorf("input file validation failed: %w", err)
 		}
```

Comment:
nit: maybe transform  these two into a private methods and call them in a single `Initialize` method. I mean, they have to be called in this order always right, so why leak this detail to a code that rely on it.

Reply:
Implemented, I think this ends up looking a bit cleaner in the test case also.